controllers/controller/devworkspacerouting/devworkspacerouting_controller.go

```diff
@@ -107,8 +107,15 @@ func (r *DevWorkspaceRoutingReconciler) Reconcile(ctx context.Context, req ctrl.
 		return reconcile.Result{}, r.finalize(solver, instance)
 	}
 
-	if instance.Annotations != nil && instance.Annotations[constants.DevWorkspaceStartedStatusAnnotation] == "false" {
-		return reconcile.Result{}, nil
+	workspaceMeta := solvers.DevWorkspaceMetadata{
+		DevWorkspaceId: instance.Spec.DevWorkspaceId,
+		Namespace:      instance.Namespace,
+		PodSelector:    instance.Spec.PodSelector,
+	}
+
+	if instance.IsWorkspaceStopped() {
+		err := solver.WorkspaceStopped(instance, workspaceMeta)
```

Comment:
Is there a case where stopping a workspace may require a second reconcile? I'm wonder if we need an error type (or to just reuse `RoutingNotReady`) to signify that a requeue is needed but no actual error occurred.